contracts/governance/Staking/Staking.sol

```diff
@@ -269,7 +269,7 @@ contract Staking is IStaking, WeightedStaking, ApprovalReceiver {
 		uint96 amount,
 		uint256 until,
 		address receiver
-	) public {
+	) public whenNotFrozen {
```

Comment:
Do we need the `whenNotFrozen` check here? The `governanceWithdraw` function can only be executed by the exchequer.

The flow looks like:
governanceWithdrawVesting(Staking) -> governanceWithdrawTokens(Vesting) -> governanceWithdraw(Staking)

Reply:
freeze function should be rarely applied - only when withdrawal is affected - exactly this case. in theory this can be vulnerable too.

Reply:
seconded. it is a failsafe measure, it has to apply everywhere to efficiently protect the contract.

Reply:
changed logic and implementation